Samsung/TizenRT
# Implement gRPC for TizenRT

Only test with **gcc-arm-none-eabi-6-2017-q1-update** [**[download]**](https://developer.arm.com/-/media/Files/downloads/gnu-rm/6_1-2017q1/gcc-arm-none-eabi-6-2017-q1-update-linux.tar.bz2?revision=6799a2bc-ea25-4e11-8a36-3f4bce1c06f7?product=GNU%20Arm%20Embedded%20Toolchain,64-bit,,Linux,6-2017-q1-update)

- Implement sendmsg/recvmsg API
- Add cxxinitialize for cpp build
- Porting protobuf to TizenRT
- Porting gRPC to TizenRT
- Add examples for protobuf client Test
- Add examples for gRPC client Test
- Add configs for gRPC build (artik053/grpc , qemu/tc_16m_grpc)
- Change SSL library (OpenSSL -> mbedTLS)


**How To Test**

0) Check ARM gcc Version (should be over 6.3)
`$ arm-none-eabi-g++ --version`

1) Use artik053/grpc
`os/tools$ ./configure.sh artik053/grpc`

2) Connect to WiFi on TASH (After flashing tinyara.bin)
`TASH>> wifi startsta`
`TASH>> wifi join [SSID] [PASSPHASE]`
`TASH>> ifconfig wl1 dhcp`

3) Run ntpclient (to sync up system time)
`TASH>> ntpclient start 600 0.asia.pool.ntp.org`
`TASH>> date`

4) Run grpc_client or route_client (before this, the grpc server should be running)
To enable secure mode, add [-s] option to the command.
`TASH>> grpc_client [-s] [server ip]:[port]`
`TASH>> route_client [-s] [server ip]:[port]`

**Example Test Environment with Linux**
1)  [TizenRT][Client] **grpc_client** <-> **greeter_server** [Server][Linux]
2) [TizenRT][Client] **route_client** <-> **route_guide_server** [Server][Linux]
